api/src/poem_backend/response.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::Display;
+
+use super::accept_type::AcceptType;
+use anyhow::format_err;
+use aptos_api_types::{LedgerInfo, U64};
+use poem::Result as PoemResult;
+use poem_openapi::{payload::Json, types::ToJSON, ApiResponse, Enum, Object, ResponseContent};
+use serde::{Deserialize, Serialize};
+
+use super::bcs_payload::Bcs;
+
+pub type AptosResult<T> = PoemResult<AptosResponse<T>, AptosErrorResponse>;
+
+// TODO: Consdider having more specific error structs for different endpoints.
+/// This is the generic struct we use for all API errors, it contains a string
+/// message and an Aptos API specific error code.
+#[derive(Debug, Object)]
+pub struct AptosError {
+    message: String,
+    error_code: Option<AptosErrorCode>,
+    aptos_ledger_version: Option<U64>,
```

Comment:
I can't wait until we can get rid of the U64 :(

Reply:
We should just write something that converts types to a JSON type if it's u64, so when we deserialize it it comes back as a u64 instead of bothering with this middle type.